src/backends/cmake.rs

```diff
@@ -22,6 +22,9 @@ impl Backend for Cmake {
     }
 
     fn build(&self, config: &BuildArgs) -> bool {
+        let create_dir =
+            fs::create_dir_all(format!("{}/build", self.lfc.out.display().to_string())).is_ok();
+
```

Comment:
This seems to be Cpp-specific. Either we have to make the backends be language-agnostic, and be extended for specific languages, or we must make them language specific. In which case we would have cmake_c.rs, cmake_cpp.rc. It seems that there are several backends that can be used for both C and Cpp. CMake, GNU Make, Meson, so the first option might be better.

Reply:
I am referring here to the build function and in general this CMake backend implementation. Not to this particular diff

Reply:
Yeah its currently CPP specific. Taking a look how the C cmake works would be one of my next points on my agenda.

Reply:
Agreed, make sense